### PR TITLE
docs(api): add `anavigateTo` docs

### PR DESCRIPTION
**Pull request by Krutie**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Added API documentation for `navigateTo` utility function

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/content/3.api/3.utils/navigate-to.md [hunk 1]
```diff
@@ -1,7 +1,68 @@
...
+
+`navigateTo` takes route as a required parameter that can be a plain string or a route object.
+
+- **route:** type: string | Route
+
+### Example: Vue component
+
+```js
+<script setup>
+    function navigate(){
```
Comment (Atinux) [inline docs/content/3.api/3.utils/navigate-to.md:19]: I think the function here is not necessary

Comment (Krutie) [reply docs/content/3.api/3.utils/navigate-to.md]: ✅ Fixed

File: docs/content/3.api/3.utils/navigate-to.md [hunk 2]
```diff
@@ -1,7 +1,66 @@
...
+
+`navigateTo` takes route as a required parameter that can be a plain string or a route object.
+
+- **route:** type: string | Route
+
+### Example: Vue component
+
+```js
+<script setup>    
+    // string
```
Comment (Atinux) [inline docs/content/3.api/3.utils/navigate-to.md:19]: I think we can remove the spaces / indentation inside the <script>

Comment (Krutie) [reply docs/content/3.api/3.utils/navigate-to.md]: ✅ Fixed

**Review by Atinux — APPROVED** [review]

File: docs/content/3.api/3.utils/navigate-to.md [hunk 3]
```diff
@@ -1,7 +1,66 @@
...
+  // set the redirect code to 301 Moved Permanently
+  return navigateTo('/search', { redirectCode: 301 })
+})
+```
+
+- **replace:** type: boolean
+
+By default, `navigateTo` pushes the given route into Vue router instance on.
+
+This behaviour can be changed by setting `replace` to `true`, to indicate that given route should be replaced.
```
Comment (DamianGlowala) [inline docs/content/3.api/3.utils/navigate-to.md:57]:
```suggestion
This behavior can be changed by setting `replace` to `true`, to indicate that given route should be replaced.
```

File: docs/content/3.api/3.utils/navigate-to.md [hunk 4]
```diff
@@ -1,7 +1,66 @@
...
+```js
+export default defineNuxtRouteMiddleware((to, from) => {
+  // set the redirect code to 301 Moved Permanently
+  return navigateTo('/search', { redirectCode: 301 })
+})
+```
+
+- **replace:** type: boolean
+
+By default, `navigateTo` pushes the given route into Vue router instance on.
```
Comment (DamianGlowala) [inline docs/content/3.api/3.utils/navigate-to.md:55]: _into Vue router instance on_ (?)

**Review by DamianGlowala — CHANGES_REQUESTED** [review]

File: docs/content/3.api/3.utils/navigate-to.md (hunk 3 above)
Comment (Krutie) [reply docs/content/3.api/3.utils/navigate-to.md]: ✅ Fixed

**Review by pi0 — APPROVED** [review]
Thanks <3

**Pull request merged**